source/Halibut/Transport/Streams/WriteIntoMemoryBufferStream.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Halibut.Transport.Observability;
+
+namespace Halibut.Transport.Streams
+{
+    public class WriteIntoMemoryBufferStream : Stream
+    {
+        readonly MemoryStream memoryStream;
+        readonly Stream sinkStream;
+        readonly long writeIntoMemoryLimitBytes;
+        readonly OnDispose onDispose;
+        bool usingSinkStream;
```

Comment:
Rename to something that suggests buffering.

Reply:
Went with `usingMemoryBuffer`, and flipped the boolean.